dvc/fs/dvc.py

```diff
@@ -251,12 +251,15 @@ def _open(
         try:
             return self.repo.fs.open(fs_path, mode=mode)
         except FileNotFoundError:
-            _, dvc_fs, subkey = self._get_subrepo_info(key)
+            repo, dvc_fs, subkey = self._get_subrepo_info(key)
             if not dvc_fs:
                 raise
 
         dvc_path = _get_dvc_path(dvc_fs, subkey)
-        return dvc_fs.open(dvc_path, mode=mode)
+        kw = {}
+        if kwargs.get("cache_remote_stream", False):
+            kw["cache_odb"] = repo.cache.local
+        return dvc_fs.open(dvc_path, mode=mode, **kw)
```

Comment:
Plots should just `dvc.fetch` with revisions they are going to be visiting so that all required data is already in cache.

Reply:
Not with `revisions` as that would involve too many passes. It still has to go through each revision to collect definitions and data (from git and dvc), etc. 

This will also require having a good filtering mechanism equivalent to what is being done in plots here (and be kept in sync with one another). 

I agree that it's how it should work, and I have been saying that for years, but on short term, this is a very simple/straightforward solution.

Reply:
@skshetry Passes can be optimized and cached for fetch, so we will be able to only check for one hash and be done with it.

> This will also require having a good filtering mechanism equivalent to what is being done in plots here (and be kept in sync with one another).

Yes, we do need that even for that studio `fetch_cache`, so I expect to have that sooner than later.

> I agree that it's how it should work ideally, and I have been saying that for years, but on short term, this is a very simple/straightforward solution.

👍 Alright, then I'm alright with merging this approach for now. We'll get back to it in the near future.

Reply:
> Yes, we do need that even for that studio fetch_cache, so I expect to have that sooner than later.

I don't know what the priority of that is. I am going to ignore Studio till then.

Regarding filtering, one thing that plots support is a directory, which could be a mix of both Git + DVC tracked files.

I am a bit worried that we would have to do two levels of filtering, one for fetch and the other when actually parsing plots, which can easily go out of sync. So even though I am not a fan of DVCFileSystem for plots parsing, it does that job really well.

Reply:
> Regarding filtering, one thing that plots support is a directory, which could be a mix of both Git + DVC tracked files.

I guess what you mean is that index.data doesn't have git files and that's true, but thats only for dvc-tracked data. We clearly need smth like index.files that won't ignoro no-cache stuff.

Reply:
yeah, and not just limited to git files. Top-level plots are by default no-cache, unless they are also recorded in stages or as outputs.

Also, at the moment, index is built from stages/outputs, so top-level plots may or may not be there. So using the present index's definition, they are not just limited to dvc's index (they might be in the workspace or in the git index).

Reply:
Anyway, you can ignore those in `fetch_cache`, and only care about dvc's index.
My point was that you'd have to use a different index when parsing plots (i.e. mixing Git + DVC's index).

Reply:
index is built from stages/outputs right now, but it is not meant to stay that way. So all the stuff that you are talking about should be in the index (we might have `data` and `files`, but that's a minor detail).